typed_grid_macro/src/with_traits.rs

```diff
@@ -2,6 +2,24 @@ use itertools::iproduct;
 use proc_macro2::*;
 use quote::quote;
 
+#[derive(Debug)]
+pub enum Direction {
+    Up,
+    Down,
+    Left,
+    Right,
+}
+
+impl Direction {
+    pub fn trait_name(&self) -> String {
+        format!("Move{:?}", self)
+    }
+
+    pub fn method_name(&self) -> String {
+        format!("{:?}", self).to_lowercase()
+    }
+}
+
```

Comment:
This might be just a bad premature generalisation and overcomplicating things, but my 2 cents: to me this looks like more broad pattern of defining algebra for your domain model, so I'd maybe call Direction an Algebra, as implementation would be pretty much the same for all algebras (if you put "Move" on enum).

Feel free to discard this comment, as Its more a way to say hi :)